backend/Application/Aggregates/Contract/Jobs/IStatelessJob.cs

```diff
@@ -0,0 +1,34 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Application.Aggregates.Contract.Jobs;
+
+/// <summary>
+/// This interface is used by jobs which should be able to run in parallel
+/// without sharing state between processes.
+/// </summary>
+public interface IStatelessJob
+{
+    /// <summary>
+    /// This returns a unique identifier of the job.
+    ///
+    /// WARNING: changing this could result in already executed jobs rerunning.
+    /// </summary>
+    string GetUniqueIdentifier();
+
+    /// <summary>
+    /// Get an enumerable with batch identifiers.
+    /// </summary>
+    /// <param name="cancellationToken"></param>
+    Task<IEnumerable<int>> GetBatches(CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Process batch.
+    /// </summary>
+    ValueTask Process(int batch, CancellationToken token = default);
```

Comment:
That is actually a nice catch!

I looked it through and my intention was to create a job which was stateless and only from some enumerable could do some work. Experience from interface `IStatelessBlockHeighJobs` is that it was to locked for general usage.

I have renamed some method parameter and documentation headers in `IStatelessJob` such that it does not say anything about batches. In this way my hope is it can be used more generically.